tests/v2/validation/rbac/etcdbackup/etcd_backup_test.go

```diff
@@ -0,0 +1,115 @@
+//go:build (validation || infra.any || cluster.any || stress) && !sanity && !extended
+
+package rbac
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/etcdsnapshot"
+	rbac "github.com/rancher/rancher/tests/framework/extensions/rbac"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ETCDRbacBackupTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (rb *ETCDRbacBackupTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *ETCDRbacBackupTestSuite) SetupSuite() {
+	rb.session = session.NewSession()
+
+	client, err := rancher.NewClient("", rb.session)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+}
+
+func (rb *ETCDRbacBackupTestSuite) testEtcdSnapshotCluster(role, name string, user *management.User) {
+
+	rb.T().Logf("Created user: %v", user.Username)
+	standardUserClient, err := rb.client.AsUser(user)
+	require.NoError(rb.T(), err)
+
+	adminProject, err := rbac.CreateProject(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+
+	if role == rbac.StandardUser {
+		if strings.Contains(role, "project") {
+			err := users.AddProjectMember(rb.client, adminProject, user, role, nil)
+			require.NoError(rb.T(), err)
+		} else {
+			err := users.AddClusterRoleToUser(rb.client, rb.cluster, user, role, nil)
+			require.NoError(rb.T(), err)
+		}
+	}
+
+	relogin, err := standardUserClient.ReLogin()
+	require.NoError(rb.T(), err)
+	standardUserClient = relogin
+
+	log.Info("Test case - Take Etcd snapshot of a cluster as a " + name)
+
+	err = etcdsnapshot.CreateRKE2K3SSnapshot(standardUserClient, rb.cluster.ID)
+	switch role {
+	case rbac.RoleClusterOwner, rbac.RestrictedAdmin:
+		require.NoError(rb.T(), err)
+
+	case rbac.RoleClusterMember, rbac.RoleProjectOwner, rbac.RoleProjectMember:
+		require.Error(rb.T(), err)
+		assert.Equal(rb.T(), "Resource type [provisioning.cattle.io.cluster] is not updatable", err.Error())
+	}
+}
+
+func (rb *ETCDRbacBackupTestSuite) TestETCDRBAC() {
```

Comment:
since the actual snapshot is taken in a private function, can you give this test a more clear name?